realm/realm-library/src/main/java/io/realm/internal/TableQuery.java

```diff
@@ -462,31 +458,19 @@ public TableView findAll() {
     // handover find* methods
     // this will use a background SharedGroup to import the query (using the handover object)
     // run the query, and return the table view to the caller SharedGroup using the handover object.
-    public long findAllWithHandover(long bgSharedGroupPtr, long nativeReplicationPtr,  long ptrQuery) throws BadVersionException {
-        validateQuery();
```

Comment:
I agree validating the query on the worker thread is a bit odd, but isn't there a chance that the query might be invalid? e.g. You move to the worker thread, but the schema has changed?

Also with this we are not validating on the UI thread in e.g. `findAllAsync()`. I don't think we will catch errors like having `beginGroup()` without `endGroup()` without it.